events_streaming.py

```diff
@@ -0,0 +1,73 @@
+from clickhouse import ClickHouse, RepoClickHouseClient
+import argparse
+from clickhub import load_config
+from repo.pull_repo import pull_data, push_data, parse_data, collect_data
+import threading
+from tests import timing
+
+
+parser = argparse.ArgumentParser(
+    description="github importer",
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+)
+
+parser.add_argument("-c", "--config", default="config.yml", help="config")
+parser.add_argument("-d", "--debug", action="store_true", help="debug")
+parser.add_argument("-s", "--size", default=10, help="batch_size")
+
+args = parser.parse_args()
+
+config = load_config(args.config)
+
+clickhouse = ClickHouse(
+    host=config["host"],
+    port=config["port"],
+    native_port=config["native_port"],
+    username=config["username"],
+    password=config["password"],
+    secure=config["secure"],
+)
+client = RepoClickHouseClient(clickhouse)
+
+# counts request rete
+count = 0
+
+
+def events_streaming(repos, lock, queue):
+    global count
+    for owner, repo in repos:
+        url = f"https://api.github.com/repos/{owner}/{repo}/events"
+        row_data, next_link, f = pull_data(url)
+        data = parse_data(row_data)
+        collect_data(lock, queue, data)
+
+        count += 1
+
+        while f:
+            row_data, next_link, f = pull_data(next_link)
+            count += 1
+            print(count)
+            data = parse_data(row_data)
+            collect_data(lock, queue, data)
+
+
+@timing.timeit
+def main():
+    repo_names = client.query_rows("SELECT DISTINCT repo_name FROM git.commits")
```

Comment:
one option is rather than have more event streaming readers, we allow the use to pass a key (0 to N) and do `repo_name % key == 0` THat way we could start N workers in different containers. We need a way to get a worker to only process a subset as one is unlikely to handle the load

Reply:
use github_events